js/codegen/main.js

```diff
@@ -73,24 +39,31 @@ editor.setOption("fontSize", "10pt");
 editor.setShowPrintMargin(false);
 editor.setTheme("ace/theme/textmate");
 editor.setWrapBehavioursEnabled(false);
-session.setMode("ace/mode/json");
+session.setMode("ace/mode/javascript");
 session.setOption("useWorker", false);
 session.setTabSize(2);
 session.setUseSoftTabs(true);
 session.setUseWrapMode(false);
 
-function onChange() {
+function onChange(codeGenerator) {
   var code = editor.getValue();
   try {
-    var ast = (0, eval)("(function(){return " + code + "}())");
-    var program = codegen.default(ast);
+    var ast = parser.parseScript(code);
```

Comment:
Let's parse it as a module instead. People would probably expect import/export declarations to work here.